middleware/protect.js

```diff
@@ -50,17 +50,31 @@ module.exports = function (keycloak, spec) {
 
   return function protect (request, response, next) {
     if (request.kauth && request.kauth.grant) {
-      if (!guard || guard(request.kauth.grant.access_token, request, response)) {
+      if (!guard) {
         return next();
+      } else {
+        if (guard.constructor.name === 'AsyncFunction') {
+          guard(request.kauth.grant.access_token, request, response).then(function (result) {
+            if (result) {
+              return next();
```

Comment:
I am not sure we need return value of `next()`, but this way `protect()` function doesn't return anything. Is it problem @abstractj @vmuzikar?

Reply:
@mhajas You mean that in case of async function we don't need to return anything? I think we still need to return `next()`, why wouldn't we?

Reply:
No I am not saying we don't need to return anything, but that we probably don't return anything when it is written as it is now. I am not sure how it works exactly with promises, but if I am not mistaken, this `return next()` doesn't belong to `protect()` function but to promise's `then()` function.

Reply:
@mhajas It seems `next()` in the `then` block really belongs to `protect`. Anyway, that's what my IDE tells me. :D

Reply:
I tried something like this:
```javascript
function a() {
  var promise = new Promise(function(resolve, reject) {
      resolve("Stuff worked!");
  });

  promise.then(function(result) {
  	console.log("then called")
    return "return from then";
  });
}

console.log("result is: " + a())
```

and it prints 

```
result is: undefined
then called
```